packages/client/src/v2-events/utils.ts

```diff
@@ -184,6 +183,47 @@ export function replacePlaceholders({
 /** Does not have parent */
 const ROOT_LOCATION_ID = '0'
 
+type LocationObject = Record<
+  'FACILITY' | 'DISTRICT' | 'STATE' | 'COUNTRY',
+  string
+>
+
+export function useResolveLocationToObject(
+  locationId: string | undefined,
+  obj: Partial<LocationObject> = {}
+): Partial<LocationObject> {
+  const locations = useSelector(getLocations)
+  const intl = useIntl()
+
+  if (!locationId) {
+    return obj
+  }
+
+  const location = locations[locationId]
+
+  if (!location) {
+    if (locationId === ROOT_LOCATION_ID) {
+      const country = countries.find(
+        (c) => c.value === window.config.COUNTRY
+      )?.label
+
+      const countryName = country ? intl.formatMessage(country) : ''
+
+      return { ...obj, COUNTRY: countryName }
+    }
+
+    return obj
+  }
+
+  const partOf = location.partOf.split('/')[1]
```

Comment:
Is this location still coming from Hearth? In the `events` database locations we have partOf field as just a plain UUID without the old `Location/` prefix